pypowerwall/tedapi/vitals_dictionary.py

```diff
@@ -0,0 +1,517 @@
+import json
+from pypowerwall import __version__
+import time
+
+def lookup(data, keylist):
+    """
+    Lookup a value in a nested dictionary or return None if not found.
+    data - nested dictionary
+    keylist - list of keys to traverse
+    """
+    for key in keylist:
+        if isinstance(data, dict):
+            data = data.get(key)
+        else:
+            return None
+    return data
+
+def calculate_dc_power(V, I):
+    power = V * I
+    return power
+
+class VitalsDictionary:
+    def __init__(self, config, status, gw_ip):
+        self.config = config
+        self.gw_ip = gw_ip
+        self.status = status
+
+    def get_vitals(self):
+        return {
+            **self.get_header(),
+            **self.get_neurio(),
+            **self.get_pvac(),
+            **self.get_pvs(),
+            **self.get_ststsm(),
+            **self.get_tepinv(),
+            **self.get_tepod(),
+            **self.get_tesla(),
+            **self.get_tesync(),
+            **self.get_tethc(),
+        }
+
+    def get_header(self):
+        header = {}
+        header["VITALS"] = {
+            "text": "Device vitals generated from Tesla Powerwall Gateway TEDAPI",
+            "timestamp": time.time(),
+            "gateway": self.gw_ip,
+            "pyPowerwall": __version__,
+        }
+        return header
+
+    def get_meters(self):
+        meter_config = {}
+        if "meters" in self.config:
+            # Loop through each meter and use device_serial as the key
+            for meter in self.config['meters']:
```

Comment:
So much nesting! Use breaks and continues to reduce nesting wherever possible. This is already done in my own branch, but it'll be awhile before that's merged.

Reply:
I'll probably leave this to you, then - I don't have this hardware, so I just lifted this from the existing file to move it here since I can't test it.

Reply:
Sounds good. I'll come in later with my refactoring :)